src/routes/v2/organizations.rs

```diff
@@ -0,0 +1,1038 @@
+            .service(delete_organization_icon)
+            .service(super::teams::team_members_get_organization),
+    );
+}
+
+#[derive(Deserialize, Validate)]
+pub struct NewOrganization {
+    #[validate(
+        length(min = 3, max = 64),
+        custom(function = "crate::util::validate::validate_name")
+    )]
+    pub name: String,
+    #[validate(length(min = 3, max = 256))]
+    pub description: String,
+    #[validate(
+        length(min = 3, max = 64),
+        regex = "crate::util::validate::RE_URL_SAFE"
+    )]
+    pub slug: String,
+    #[serde(default = "crate::models::teams::ProjectPermissions::default")]
+    pub default_project_permissions: ProjectPermissions,
+    #[validate(custom(function = "crate::util::validate::validate_urls"))]
+    /// Any desired links for the organization.
+    /// ie: "discord" -> "https://discord.gg/..."
+    #[serde(default)]
+    pub urls: HashMap<String, String>,
+}
+
+#[post("organization")]
+pub async fn organization_create(
+    req: HttpRequest,
+    new_organization: web::Json<NewOrganization>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, CreateError> {
+    let current_user = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::ORGANIZATION_CREATE]),
+    )
+    .await?
+    .1;
+
+    new_organization
+        .validate()
+        .map_err(|err| CreateError::ValidationError(validation_errors_to_string(err, None)))?;
+
+    let mut transaction = pool.begin().await?;
+
+    // Try slug
+    let slug_organization_id_option: Option<OrganizationId> =
+        serde_json::from_str(&format!("\"{}\"", new_organization.slug)).ok();
+
+    if let Some(slug_organization_id) = slug_organization_id_option {
+        let slug_organization_id: models::ids::OrganizationId = slug_organization_id;
+        let results = sqlx::query!(
+            "
+            SELECT EXISTS(SELECT 1 FROM organizations WHERE id=$1)
+            ",
+            slug_organization_id.0 as i64
+        )
+        .fetch_one(&mut *transaction)
+        .await
+        .map_err(|e| CreateError::DatabaseError(e.into()))?;
+
+        if results.exists.unwrap_or(false) {
+            return Err(CreateError::SlugCollision);
+        }
+    }
+
+    {
+        let results = sqlx::query!(
+            "
+            SELECT EXISTS(SELECT 1 FROM organizations WHERE slug = LOWER($1))
+            ",
+            new_organization.slug
+        )
+        .fetch_one(&mut *transaction)
+        .await
+        .map_err(|e| CreateError::DatabaseError(e.into()))?;
+
+        if results.exists.unwrap_or(false) {
+            return Err(CreateError::SlugCollision);
+        }
+    }
+
+    // Gets the user database data
+    let user_data =
+        crate::database::models::User::get_id(current_user.id.into(), &mut transaction, &redis)
```

Comment:
why is this query needed? you already have the user

Reply:
I was using it for the db Team Member -> models Team Member function which requires db Users and not models Users. I can create a version of that function that uses the models version though as it looks like that doesnt use the db version for anything other than converting it